game/groundforces/template_loader.py

```diff
@@ -0,0 +1,361 @@
+from __future__ import annotations
+
+import itertools
+import logging
+import pickle
+from concurrent.futures import ThreadPoolExecutor
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import Iterator, Optional, Any
+
+import dcs
+import yaml
+from dcs import Point
+from dcs.unitgroup import StaticGroup
+
+from game import persistency
+from game.version import VERSION
+from game.data.groups import GroupRole, GroupTask
+from game.data.units import UnitClass
+from game.groundforces.template import (
+    GroundObjectTemplate,
+    GroupTemplate,
+    UnitTemplate,
+    AntiAirTemplate,
+    BuildingTemplate,
+    NavalTemplate,
+    GroundForceTemplate,
+    DefensesTemplate,
+)
+from game.profiling import logged_duration
+
+TEMPLATE_DIR = "resources/templates/"
+TEMPLATE_DUMP = "Liberation/templates.p"
+
+
+@dataclass
+class GroupTemplateMapping:
+    # The group name used in the template.miz
+    name: str
+
+    # Defines if the group is required for the template or can be skipped
+    optional: bool = False
+
+    # All static units for the group
+    statics: list[str] = field(default_factory=list)
+
+    # Defines to which tgo group the groupTemplate will be added
+    # This allows to merge groups back together. Default: Merge all to group 1
+    group: int = field(default=1)
+
+    # Randomization settings. If left empty the template will be generated with the
+    # exact values (amount of units and unit_type) defined in the template.miz
+    # How many units the template should generate. Will be used for randomization
+    unit_count: list[int] = field(default_factory=list)
+    # All unit types the template supports.
```

Comment:
So if this is empty, any units that fit the role are allowed, but if non-empty, _only_ those units are valid? Comment should elaborate.